faker/utils/loading.py

```diff
@@ -4,6 +4,13 @@
 
 def list_module(module):
     path = os.path.dirname(module.__file__)
+    try:
+        import pkgutil
+        modules = [name for finder, name, is_pkg in pkgutil.iter_modules([path]) if is_pkg]
+        if len(modules) > 0:
+            return modules
+    except (ImportError, AttributeError):
+        pass
```

Comment:
pkgutil.iter_modules does work for non-zip packages.  But there does seem
to be a bug with it in python 3.2.5 where the bootstrapping File Loader is
missing an implementation. So, the fallback code is still required to get
the correct behavior on python 3.2.5 when using the standard file loader.

On Wed, Mar 18, 2015 at 7:13 AM, Flavio Curella <notifications@github.com>
wrote:

> In faker/utils/loading.py
> <https://github.com/joke2k/faker/pull/212#discussion_r26665749>:
>
> > @@ -4,6 +4,13 @@
> >
> >  def list_module(module):
> >      path = os.path.dirname(module.__file__)
> > +    try:
> > +        import pkgutil
> > +        modules = [name for finder, name, is_pkg in pkgutil.iter_modules([path]) if is_pkg]
> > +        if len(modules) > 0:
> > +            return modules
> > +    except (ImportError, AttributeError):
> > +        pass
>
> I'd rather avoid an except ...: pass pattern. Does pkgutil.iter_modules
> work for non-zip packages as well? If it does and it's available on all
> Python versions that we support, I'd rather just use that as the one single
> way to list modules.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joke2k/faker/pull/212/files#r26665749>.
>